### PR TITLE
add cache provision

### PR DESCRIPTION
**Pull request by Ghostbaby**
Signed-off-by: Ghostbaby zhuhuijunzhj@gmail.com
fix https://github.com/goharbor/harbor-cluster-operator/issues/24

### CODE REVIEW

File: controllers/cache/generate.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+			Namespace: redis.Namespace,
+			Labels:    labels,
+		},
+		StringData: map[string]string{
+			"password": passStr,
+		},
+	}
+}
+
+func generateRedisStorage(size, name string, labels map[string]string) *corev1.PersistentVolumeClaim {
```
Comment (wangcanfengxs) [inline controllers/cache/generate.go:88]: why this function receiver is not `*RedisReconciler`？

File: controllers/cache/labels.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package cache
+
+const (
+	AppLabel = "harbor-cluster-operator"
+)
+
+// NewLabels returns new labels
+func (redis *RedisReconciler) NewLabels() map[string]string {
+	dynLabels := map[string]string{
+		AppLabel:                       redis.Name,
```
Comment (wangcanfengxs) [inline controllers/cache/labels.go:10]: If we need to add some labels on component CRs, we should extract some const variable about the labels name.

File: controllers/cache/redis.go [hunk 3]
```diff
@@ -8,16 +8,47 @@ import (
...
+	crStatus, err := redis.Provision(crStatus)
+	if err != nil {
+		return crStatus, err
+	}
 	return nil, nil
 }
 
-func (redis *RedisReconciler) Provision(spec *goharborv1.HarborCluster) (*lcm.CRStatus, error) {
-	panic("implement me")
+func (redis *RedisReconciler) Provision(crStatus *lcm.CRStatus) (*lcm.CRStatus, error) {
```
Comment (wangcanfengxs) [inline controllers/cache/redis.go:41]: This is a interface method implement. MUST NOT change the signature of method.

Comment (Ghostbaby) [reply controllers/cache/redis.go:41]: ok, I will move CRstatus to RedisReconciler.

**Pull request closed without merging**